beacon-chain/sync/rpc_beacon_blocks_by_root.go

```diff
@@ -25,6 +25,7 @@ func (s *Service) sendRecentBeaconBlocksRequest(ctx context.Context, blockRoots
 		}
 		s.pendingQueueLock.Lock()
 		if err := s.insertBlockToPendingQueue(blk.Block().Slot(), blk, blkRoot); err != nil {
+			s.pendingQueueLock.Unlock()
 			return err
 		}
 		s.pendingQueueLock.Unlock()
```

Comment:
```suggestion
		s.pendingQueueLock.Lock()
		defer s.pendingQueueLock.Unlock()
		if err := s.insertBlockToPendingQueue(blk.Block().Slot(), blk, blkRoot); err != nil {
			return err
		}
```
This is a better pattern for you.

Reply:
Thanks @saolyn, it's ready again!